### ISSUE TITLE
ECS clusters: enable Container Insights

### ISSUE THREAD

**Issue by domcleal** [terraform-providers/terraform-provider-aws#9294]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

ECS clusters have an optional [Container Insights feature](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights.html), which can be enabled on a per-cluster basis in CreateCluster API call.

This enables more CloudWatch metrics to be recorded about the running services and tasks. It should be possible to enable this in `aws_ecs_cluster` resources.

### New or Affected Resource(s)

* `aws_ecs_cluster`

### Potential Terraform Configuration

```hcl
resource "aws_ecs_cluster" "foo" {
  name               = "white-hart"
  container_insights = true
}
```

### References

* https://aws.amazon.com/about-aws/whats-new/2019/07/introducing-container-insights-for-ecs-and-aws-fargate-in-preview/
* [docs.aws.amazon.com: Setting Up Container Insights on Amazon ECS](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/deploy-container-insights-ECS.html)
* [API: CreateCluster settings](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateCluster.html)

**Comment by sunilkumarmohanty** [comment]
Needs v1.21.0 aws sdk

**Comment by chaspy** [comment]
Hi, let me try

**Comment by ewbankkit** [comment]
Amazon CloudWatch Container Insights is now GA - [Blog post](https://aws.amazon.com/blogs/aws/operational-insights-for-containers-and-containerized-applications/).

**Comment by bflad** [comment]
Support for a new `setting` configuration block in the `aws_ecs_cluster` resource has been merged and will release with version 2.27.0 of the Terraform AWS Provider, tomorrow. 👍 Thanks to @chaspy and @stack72 for their implementation work.
